python/private/get_local_runtime_info.py

```diff
@@ -13,9 +13,158 @@
+  # on the system - some systems like version suffix, others don't.""
+  #
+  # A typical INSTSONAME is 'libpython3.8.so.1.0' on Linux, or
+  # 'Python.framework/Versions/3.9/Python' on MacOS. Due to the possible
+  # version suffix we have to find the suffix within the filename.
+  #
+  # A typical LDLIBRARY is 'libpythonX.Y.so' on Linux, or 'pythonXY.dll' on
+  # Windows.
+  #
+  # A typical LIBRARY is 'libpythonX.Y.a' on Linux.
+  lib_names = [
+      get_config(x)
+      for x in (
+          "INSTSONAME",
+          "LDLIBRARY",
+          "PY3LIBRARY",
+          "LIBRARY",
+          "DLLLIBRARY",
+      )
+  ]
+
+  if _IS_WINDOWS:
+    so_prefix = ""
+  else:
+    so_prefix = "lib"
+
+  # Get/override the SHLIB_SUFFIX, which is typically ".so" on Linux and
+  # ".dylib" on macOS.
+  so_suffix = get_config("SHLIB_SUFFIX")
+  if sys.platform == "darwin":
+    so_suffix = ".dylib"
+  elif _IS_WINDOWS:
+    so_suffix = ".lib"
+  elif not so_suffix:
+    so_suffix = ".so"
+
+  version = get_config("VERSION")
+  abiflags = get_config("ABIFLAGS") or get_config("abiflags") or ""
+
+  # On Windows, extensions should link with the pythonXY.lib files.
+  # See: https://docs.python.org/3/extending/windows.html
+  # So ensure that the pythonXY.lib files are included in the search.
+  if abiflags:
+    lib_names.append(f"{so_prefix}python{version}{abiflags}{so_suffix}")
+  lib_names.append(f"{so_prefix}python{version}{so_suffix}")
+
+  # Dedup and remove empty values, keeping the order.
+  lib_names = [v for v in lib_names if v]
+  return {k: None for k in lib_names}.keys()
+
+
+def _get_python_library_info():
+  """Returns a dictionary with the static and dynamic python libraries."""
+  config_vars = sysconfig.get_config_vars()
+
+  # VERSION is X.Y in Linux/macOS and XY in Windows.
+  if not config_vars.get("VERSION"):
+    if sys.platform == "win32":
+      config_vars["VERSION"] = (
+          f"{sys.version_info.major}{sys.version_info.minor}"
+      )
+    else:
+      config_vars["VERSION"] = (
+          f"{sys.version_info.major}.{sys.version_info.minor}"
+      )
+
+  search_directories = _search_directories(config_vars.get)
+  search_libnames = _search_library_names(config_vars.get)
+
+  dynamic_libraries = {}
+  static_libraries = {}
+  for root_dir in search_directories:
+    for libname in search_libnames:
+      composed_path = os.path.join(root_dir, libname)
+      if not os.path.exists(composed_path) or os.path.isdir(composed_path):
+        continue
+      if composed_path.endswith(".lib") or composed_path.endswith(".a"):
+        static_libraries[composed_path] = None
+      else:
+        dynamic_libraries[composed_path] = None
+
+  # NOTE: It's possible that including the dynamic libraries currently loaded
+  # by the running python interpreter would be a useful addition.
+
+  return {
+      "dynamic_libraries": list(dynamic_libraries.keys()),
+      "static_libraries": list(static_libraries.keys()),
+  }
```

Comment:
There is a mix of 2 and 4 spaces here. Could you please ensure that you have run `pre-commit run black` on this file? You could also install the pre-commit hooks with `pre-commit install`.

Reply:
Looks like this was done; resolving